src/dotnet-sql-cache/SqlQueries.cs

```diff
@@ -8,8 +8,10 @@ namespace Microsoft.Extensions.Caching.SqlConfig
     internal class SqlQueries
     {
         private const string CreateTableFormat = "CREATE TABLE {0}(" +
-            // add collation to the key column to make it case-sensitive
-            "Id nvarchar(900) COLLATE SQL_Latin1_General_CP1_CS_AS NOT NULL, " +
+            // - Primary key column cannot exceed the maximum size of 900 bytes. Here each character occupies
+            //   2 bytes.
+            // - Add collation to the key column to make it case-sensitive
+            "Id nvarchar(445) COLLATE SQL_Latin1_General_CP1_CS_AS NOT NULL, " +
```

Comment:
> do we know what happens when you do?
I noticed that if a key greater than 898 bytes is provided, then the value is truncated to 898.